src/CommonGLPI.php

```diff
@@ -710,14 +710,30 @@ private static function getTabIconClass(?string $form_itemtype = null): string
      * icon to be displayed, unless you manually specify the icon.
      *
      * @param string  $text text to display
-     * @param integer $nb   number of items (default 0)
+     * @param callable|int $nb   Callable that returns the number of items. The callable will be evaluated only if the user preference is set to display the counter.
+     *                           A zero may be directly passed if no counter is needed.
      * @param class-string<CommonGLPI>|null $form_itemtype
      * @param string $icon
      *
      *  @return string The tab text (including icon and counter if applicable)
      **/
-    public static function createTabEntry($text, $nb = 0, ?string $form_itemtype = null, string $icon = '')
+    public static function createTabEntry($text, int|callable $nb = 0, ?string $form_itemtype = null, string $icon = ''): string
     {
+        if (is_callable($nb)) {
+            if ($_SESSION['glpishow_count_on_tabs']) {
+                $nb = $nb();
+            } else {
+                $nb = 0;
+            }
+        } else if ($nb > 0) {
```

Comment:
Why not checking with is_numeric here and support also the checking directly for this case about glpishow_count_on_tabs ?